controlpanel/api/aws.py

```diff
@@ -23,11 +23,22 @@ def s3_arn(resource):
     return arn("s3", resource)
 
 
-def iam_arn(resource, account=settings.AWS_ACCOUNT_ID):
+def iam_arn(resource, account=settings.AWS_DATA_ACCOUNT_ID):
     return arn("iam", resource, account=account)
 
 
-BASE_ROLE_POLICY = {
+def iam_assume_role_principal():
+    cross_account = settings.AWS_COMPUTE_ACCOUNT_ID != settings.AWS_DATA_ACCOUNT_ID
+
+    # Principal must be source AWS account ID (not full ARN) in cross-account setup
+    if cross_account:
+        return settings.AWS_COMPUTE_ACCOUNT_ID
+
+    # Principal is full ARN of assuming IAM role in single-account setup
+    return iam_arn(f"role/{settings.K8S_WORKER_ROLE_NAME}")
+
+
+BASE_ASSUME_ROLE_POLICY = {
```

Comment:
FYI: I've renamed this slightly to be more explicit about the fact this is not a "normal" IAM policy but instead is the "Assume Role Policy" (AKA "trust relationship" policy)